app/src/main/java/com/example/breakoutgame/GameLoop.java

```diff
@@ -10,94 +10,48 @@ public class GameLoop extends Thread{
     private boolean isRunning = false;
     private SurfaceHolder surfaceHolder;
     private Game game;
-    private double averageUps;
-    private double averageFps;
-    private static final double MAX_UPS = 60.0;
-    private static final double UPS_Period = 1E+3/MAX_UPS;
 
     public GameLoop(Game game, SurfaceHolder surfaceHolder) {
         this.game = game;
         this.surfaceHolder = surfaceHolder;
     }
 
-    public double getAverageUps() {
-        return averageUps;
-    }
-
-    public double getAverageFPS() {
-        return averageFps;
-    }
-
     @Override
     public void run() {
         super.run();
 
-        //Declare time and cycle count variables
-        int updateCount = 0;
-        int frameCount = 0;
-
-        long startTime;
-        long elapsedTime;
-        long sleepTime;
-
         //Game Loop
         Canvas canvas = null;
-        startTime = System.currentTimeMillis();
         while (isRunning){
-
             //Try to update and render game
             try {
                 canvas = surfaceHolder.lockCanvas();
-                synchronized (surfaceHolder){
-                    game.update();
-                    updateCount ++;
-                    game.draw(canvas);
+                synchronized (this){
+                    if (game.playerLost()) {
+                        game.initialScreen(canvas);
+                    } else if (game.proceedToNextLevel()){
+                        game.nextLevelGeneration();
+                    } else{
```

Comment:
last else doesnt make sense? We are checking if player won or he lost, what's the third situation where we're using this else for?

Reply:
If its mandatory thingy, place it in `finally`